pkg/domain/infra/abi/play.go

```diff
@@ -964,3 +978,33 @@ func (ic *ContainerEngine) PlayKubeDown(ctx context.Context, body io.Reader, _ e
 
 	return reports, nil
 }
+
+// playKubeSecret allows users to create and store a kubernetes secret as a podman secret
+func (ic *ContainerEngine) playKubeSecret(secret *v1.Secret) (*entities.SecretCreateReport, error) {
+	r := &entities.SecretCreateReport{}
+
+	// Create the secret manager before hand
+	secretsManager, err := ic.Libpod.SecretsManager()
+	if err != nil {
+		return nil, err
+	}
+
+	// only store the data!!! if we store the entire secret marshaled,
+	// we cannot unmarshal it later. (yaml.unmarshal does not like string --> byte conversion)
+	data, err := yaml.Marshal(secret.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsPath := ic.Libpod.GetSecretsStorageDir()
+	opts := make(map[string]string)
+	opts["path"] = filepath.Join(secretsPath, "filedriver")
+	secretID, err := secretsManager.Store(secret.Name, data, "file", opts)
```

Comment:
I imagine the `name` of the secret to be the _hash_/_digest_ of the secret struct.  The risk I see with using `secret.Name` is that it's not unique. By the laws of copy-pasting, I imagine many K8s yamls to use the same secret name.

Reply:
We're doing something similar with the "service" containers of a K8s yaml (see https://github.com/containers/podman/blob/main/pkg/domain/infra/abi/play.go#L89).

Reply:
good point, I will change this around to use a hashing function

Reply:
I think they are two separate things.  The hash would just be used internally by Podman to store and lookup the secret(s).